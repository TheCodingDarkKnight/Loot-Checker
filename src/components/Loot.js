import React, { Component } from "react";
import { connect } from "react-redux";

import { fetchBitcoin } from "../actions/bitcoin";

export class Loot extends Component {
    state = {
        balance: 0
    }

    componentDidMount() {
        if (this.props.fetchBitcoin) {
            this.props.fetchBitcoin()
        }
    }

    computedBitcoin = () => {
        const { bitcoin } = this.props;

        if (!bitcoin || Object.keys(bitcoin).length === 0) return '';

        return this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''), 10);
    }
    render() {
        return (<div>
            <h3 className="balance">Bitcoin balance: {this.computedBitcoin()}</h3>
        </div>)
    }
}

export default connect(state => ({ state: state }), { fetchBitcoin })(Loot);