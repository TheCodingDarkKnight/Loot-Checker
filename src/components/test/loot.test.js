import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure, shallow } from 'enzyme';

import { Loot } from "../Loot";

configure({ adapter: new Adapter() });

describe("Loot", () => {
    const mockFetchBitcoin = jest.fn();
    let props = { balance: 20, bitcoin: {} }
    let loot = shallow(<Loot {...props} />);

    it("Renders properly", () => {
        expect(loot).toMatchSnapshot();
    });

    describe('when mounted', () => {
        beforeEach(() => {
            props.fetchBitcoin = mockFetchBitcoin;
            loot = mount(<Loot {...props} />);
        });

        it("Dispatches the `fetchBitcoin()` it recieves from props", () => {
            expect(mockFetchBitcoin).toHaveBeenCalled();
        })
    })

    describe('When there are valid bitcoin props', () => {
        beforeEach(() => {
            props = { balance: 10, bitcoin: { bpi: { USD: { rate: '1,000' } } } }
            loot = shallow(<Loot {...props} />);
        })

        it("Displays correct bitcoin value", () => {
            expect(loot.find("h3").text()).toEqual("Bitcoin balance: 0.01");
        })
    })
})

