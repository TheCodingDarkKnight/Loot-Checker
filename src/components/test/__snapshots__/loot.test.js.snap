// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot Renders properly 1`] = `ShallowWrapper {}`;
